### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by Incurso**
Added tests for game logic and implemented the basic game logic.

### CODE REVIEW

File: src/logic/tictactoe.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    for (let i = 0; i < 3; i++) {
+      let row = []
+      for (let j = 0; j < 3; j++) {
+        row.push(this.emptyToken)
+      }
+      this.board.rows.push(row)
+    }
+  }
+
+  makeMove (c) {
```
Comment (styrmird) [inline src/logic/tictactoe.js:32]: c for char? Perhaps a more descriptive name would be clearer?

File: src/logic/tictactoe.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  newGame () {
+    // Reset the board
+    this.board = {
+      rows: [],
+      moves: 0,
+      maxMoves: 9,
+      currentPlayer: 'X'
+    }
+
+    // Fill the board with emptyTokens
```
Comment (styrmird) [inline src/logic/tictactoe.js:22]: Would be simpler to initialize the board all at once i.e. something like
`board = [['-', '-', '-'], ['-', '-', '-'], ['-', '-', '-']]` 
or
`board = [[emptyToken, emptyToken, emptyToken], [emptyToken, emptyToken, emptyToken], [emptyToken, emptyToken, emptyToken]]`

?

File: src/logic/tictactoe.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      }
+      this.board.rows.push(row)
+    }
+  }
+
+  makeMove (c) {
+    let board = this.board // This works because Objects are passed by reference and not by value
+
+    // Check if the move is valid, return false if move is invalid
+    if (board.rows[c[0]][c[1]] !== this.emptyToken ||
```
Comment (styrmird) [inline src/logic/tictactoe.js:36]: Is there a more legible way to dereference the `c` argument ?

**Review by styrmird — APPROVED** [review]
Just a few questions/ ideas regarding code clarity but nothing critical

**Review by DieterMut — COMMENTED** [review]
Too many tests in one run, and no program code that uses it. It would be better to write a test and the accompanying code in the program, and commit, then taking the next step, i.e. test-code pair.

File: src/logic/tictactoe.js (hunk 1 above)
Comment (Incurso) [reply src/logic/tictactoe.js:32]: c is for cell, and yes clearly it can be better named.

**Comment by Incurso** [comment]
Way to much implemented with tests without committing.
Closing to rework.

**Pull request closed without merging**